at_functional_test/test/atclient_sync_callback_test.dart

```diff
@@ -29,6 +31,86 @@ void main() {
     }
     await Future.delayed(Duration(seconds: 10));
   });
+
+  test('delete of a key to sharedWith atSign - using await', () async {
+    AtSignLogger.root_level = 'finest';
+    final atSign = '@alice🛠';
+    var preference = TestUtils.getPreference(atSign);
+    var atClientManager = await AtClientManager.getInstance()
+        .setCurrentAtSign(atSign, 'wavi', TestUtils.getPreference(atSign));
+    final progressListener = MySyncProgressListener();
+    atClientManager.syncService.addProgressListener(progressListener);
```

Comment:
can we have separate SyncProgressListener for each test..
inside each listener you can check for exact keys that got synced